pkg/common/k8s.go

```diff
@@ -150,3 +151,21 @@ func SupportedMetricsAPIVersionAvailable(discoveredAPIGroups *metav1.APIGroupLis
 	}
 	return false
 }
+
+func IsRunningOnOpenShift(discoveredAPIGroups *metav1.APIGroupList) bool {
+	for _, group := range discoveredAPIGroups.Groups {
+		if group.Name == "config.openshift.io" {
+			return true
+		}
+	}
+	return false
+}
+
+func GetOpenShiftVersion(c k8sClient.Client) (string, error) {
+	cv := &configv1.ClusterVersion{}
+	err := c.Get(context.Background(), k8sClient.ObjectKey{Name: "version"}, cv)
+	if err != nil {
+		return "", err
+	}
+	return cv.Status.Desired.Version, nil
```

Comment:
is there not a current version?  This will mark it as the intended version during upgrade i'm guessing